test/patest_wasapi_ac3.c

```diff
@@ -0,0 +1,206 @@
+/** @file patest_wasapi_ac3.c
+    @ingroup test_src
+    @brief Use WASAPI-specific interface to send raw AC3 data to a S/PDIF output.
+    @author Ross Bencina <rossb@audiomulch.com>, Jie Ding <gabys999@gmail.com>
+*/
+/*
+ * $Id: $
+ * Portable Audio I/O Library
+ * WASAPI ac3 sound output test
+ *
+ * Copyright (c) 2009-2023 Ross Bencina, Jie Ding
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining
+ * a copy of this software and associated documentation files
+ * (the "Software"), to deal in the Software without restriction,
+ * including without limitation the rights to use, copy, modify, merge,
+ * publish, distribute, sublicense, and/or sell copies of the Software,
+ * and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR
+ * ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+ * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/*
+ * The text above constitutes the entire PortAudio license; however,
+ * the PortAudio community also makes the following non-binding requests:
+ *
+ * Any person wishing to distribute modifications to the Software is
+ * requested to send the modifications to the original developer so that
+ * they can be incorporated into the canonical version. It is also
+ * requested that these non-binding requests be included along with the
+ * license above.
+ */
+
+#include <stdio.h>
+
+#include <windows.h>    /* required when using pa_win_wasapi.h */
+
+#include "portaudio.h"
+#include "pa_win_wasapi.h"
+
+#define NUM_SECONDS         (200)
+#define SAMPLE_RATE         (48000)
+#define FRAMES_PER_BUFFER   (64)
+#define CHANNEL_COUNT       (2)
+
+#define AC3_FILEPATH        "./test_48k.ac3.spdif"
+
+typedef struct
+{
+    short *buffer;
+    int bufferSampleCount;
```

Comment:
```suggestion
    int bufferFrameCount;
```
Count whole frames, not 2-byte words.